src/test/java/org/opentripplanner/routing/core/AccessibilityRoutingTest.java

```diff
@@ -0,0 +1,304 @@
+
+        assertEquals(1, transitLegs.size());
+
+        Leg leg = transitLegs.get(0);
+        assertEquals("BLUE", leg.routeShortName);
+
+        assertEquals("GWCC-CNN CENTER STATION", leg.from.name);
+        assertEquals("ASHBY STATION", leg.to.name);
+
+        // the start has unknown access, so we get a lowered score
+        assertEquals(0.833333, leg.accessibilityScore, DELTA);
+
+
+        // if we increase the walking distance and the penalty another stop is used instead
+
+        i = getTripPlan(start, end, r -> {
+            r.unknownWheelchairAccessAtStopPenalty = 10000;
+        }).itinerary.get(0);
+        transitLegs = i.legs.stream().filter(Leg::isTransitLeg).collect(Collectors.toList());
+
+        leg = transitLegs.get(0);
+        assertEquals("BLUE", leg.routeShortName);
+
+        assertEquals("FIVE POINTS STATION", leg.from.name);
+        assertEquals("ASHBY STATION", leg.to.name);
+
+        // because we are walking to a station with bad accessibility we get a lower score
+        assertEquals(0.666f, leg.accessibilityScore, DELTA);
+    }
+
+    @Test
+    public void canScoreTripWithInaccessibleTransitLeg() {
+        // near Georgia State station
+        GenericLocation start = new GenericLocation(33.75054, -84.38409);
+        // near Ashby station
+        GenericLocation end = new GenericLocation(33.75744, -84.41754);
+
+        // we are banning the blue line even though it is accessible to force getting on the green one
+        Itinerary i = getTripPlan(start, end, r -> r.setBannedRoutes("MARTA_BLUE")).itinerary.get(0);
+        List<Leg> transitLegs = i.legs.stream().filter(Leg::isTransitLeg).collect(Collectors.toList());
+
+        Leg leg = transitLegs.get(0);
+        assertEquals("GREEN", leg.routeShortName);
+
+        assertEquals("GEORGIA STATE STATION", leg.from.name);
+        assertEquals("ASHBY STATION", leg.to.name);
+
+        // because are using a trip that has unknown accessibility, it has a lower score
+        assertEquals(0.8333f, leg.accessibilityScore, DELTA);
+    }
+
+    @Test
+    public void canApplyStairsPenalty() {
+        GenericLocation start = new GenericLocation(33.75630, -84.39527);
+        GenericLocation end = new GenericLocation(33.75649, -84.39580);
+
+        Itinerary i = getTripPlan(start, end, r -> r.setMode(TraverseMode.WALK)).itinerary.get(0);
+        Leg leg = i.legs.get(0);
+        assertEquals("WALK", leg.mode);
+        assertThatPolylinesAreEqual("y_`mEnmbbObA}@U]{@wAIMEOMHwBz@^p@Zn@Zx@Ld@DJLd@?@", leg.legGeometry.getPoints());
+
+        // if we reduce the stair penalty for wheelchairs we get a route that goes up the stairs
+        i = getTripPlan(start, end, r -> {
+            r.setMode(TraverseMode.WALK);
+            r.wheelchairStairsReluctance = 0;
+        }).itinerary.get(0);
+
+        leg = i.legs.get(0);
+        assertEquals("WALK", leg.mode);
+        assertThatPolylinesAreEqual("y_`mEnmbbO[XIHBFUTLd@?@", leg.legGeometry.getPoints());
```

Comment:
This test could use some assertions that checks for a correctly-calculated accessibility score derived from stairs influencing the walk leg score.